src/ID3Writer.mjs

```diff
@@ -18,7 +18,12 @@ import {
 
 export default class ID3Writer {
   _setIntegerFrame(name, value) {
-    const integer = parseInt(value, 10);
+    let integer;
+    if (name === 'TDAT') {
+      integer = String(parseInt(value, 10)).padStart(4, '0');
```

Comment:
![9233-hide-the-pain-harold](https://user-images.githubusercontent.com/2605831/228800495-3ed432e6-5e06-4a27-bf02-cf1e72638bb5.png)

Reply:
I'll think how to avoid such pain..

Reply:
Looks like this frame should not be integer frame, but string..

Reply:
Look at https://github.com/egoroof/browser-id3-writer/pull/83

Reply:
LGTM